sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Configs.java

```diff
@@ -102,7 +104,11 @@ public static int getCPUCnt() {
     private SslContext sslContextInit() {
         try {
             SslProvider sslProvider = SslContext.defaultClientProvider();
-            return SslContextBuilder.forClient().sslProvider(sslProvider).build();
+            SslContextBuilder sslContextBuilder = SslContextBuilder.forClient();
+            if (getJVMConfigAsBoolean(SKIP_SSL_CHECK, false)) {
+                sslContextBuilder.trustManager(InsecureTrustManagerFactory.INSTANCE);
```

Comment:
This simplifies the testing with Cosmos Emulator. However the reason we didn't do this was the concern around security if the end user mistakenly disables this in the prod environment.

IMO it is good to have this @kirankumarkolli for across SDK discussion.

Reply:
I see your point but current state of testing is unfortunately too problematic. There is not any easy way to fix this and I think such libraries should consider to test against testing environments and emulators as well. Saving certificates for each run of certificates is not an option. 

It is pretty common to disabling such things in the first place. There are always such pros and cons. Also having enabled this on production does not mean a security problem unless there is an attack against certificates. 

However if you want I can add this parameter to `builder` but I have not seen any builder methods to configuring `Configs`. This sounds for me more appropriate though.

Reply:
I want to open a PR against `TestContainers` project too but using hacky things such as reflection to configure it will be a big deal and it will be rejected at first place.

```
  private CosmosClientBuilder newSslTrustAllBuilder() {
    try {
      CosmosClientBuilder builder = new CosmosClientBuilder();
      //
      Field configs = CosmosClientBuilder.class.getDeclaredField("configs");
      configs.setAccessible(true);
      Configs realConfigs = (Configs) configs.get(builder);
      //
      Field configSslContext = Configs.class.getDeclaredField("sslContext");
      configSslContext.setAccessible(true);
      //
      SslContext newSslContext = SslContextBuilder.forClient()
                                                  .trustManager(InsecureTrustManagerFactory.INSTANCE)
                                                  .build();
      configSslContext.set(realConfigs, newSslContext);
      //
      return builder;
    } catch (Exception ex) {
      throw new IllegalStateException(ex);
    }
  }
```